### PR TITLE
Recompback

### PR DESCRIPTION
**Pull request by pent0**
TODO:
- [x] Literals
- [x] f16/f32 pack/unpack/load/store
- [x] ~~Revisit VPCK decoding (there is something wrong)~~ There is nothing wrong with it.
- [x] Phase. Secondary program goes first
- Instructions:
   * [x] VPCK (f16 + f32)
   * [x] VFRC (f16 + f32)

TODO next PR:
- Block support: Branch and recompile
- Constant table (*C* bank)
- f16/f32 uniform/buffer parameter (?)

### CODE REVIEW

File: src/emulator/shader/src/spirv_recompiler.cpp [hunk 1]
```diff
@@ -754,8 +754,8 @@ static SpirvShaderParameters create_parameters(spv::Builder &b, const SceGxmProg
     }
 
     // Create temp reg vars
-    for (auto i = 0; i < program.temp_reg_count1; i++) {
-        auto name = fmt::format("r{}", i);
+    for (auto i = 0; i < (program.temp_reg_count1 / 4) + ((program.temp_reg_count1 % 4) != 0 ? 1 : 0); i++) {
```
Comment (VelocityRa) [inline src/emulator/shader/src/spirv_recompiler.cpp]: extract this to locals above please

**Comment by pent0** [comment]
The last commit may got dropped.

Please tests. It may stop crash other games. Games with f16 as input/output/uniform parameter won't work just yet, it may displays an annoying purple color

File: src/emulator/shader/src/usse_translator_entry.cpp [hunk 2]
```diff
@@ -235,21 +236,27 @@ void convert_gxp_usse_to_spirv(spv::Builder &b, const SceGxmProgram &program, co
 
     std::map<ShaderPhase, std::vector<uint64_t>> shader_code;
 
-    // Collect insructions of Pixel (primary) phase
+    // Collect insructions of Pixel (secondary) phase
```
Comment (VelocityRa) [inline src/emulator/shader/src/usse_translator_entry.cpp]: It runs second, but it's still called primary

File: src/emulator/shader/src/usse_translator_entry.cpp [hunk 3]
```diff
@@ -235,21 +236,27 @@ void convert_gxp_usse_to_spirv(spv::Builder &b, const SceGxmProgram &program, co
...
     for (auto instr_idx = 0; instr_idx < primary_program_instr_count; ++instr_idx)
         shader_code[ShaderPhase::Pixel].push_back(primary_program[instr_idx]);
 
-    // Collect insructions of Sample rate (secondary) phase
+    // Collect insructions of Sample rate (primary) phase
```
Comment (VelocityRa) [inline src/emulator/shader/src/usse_translator_entry.cpp]: It runs first, but it's still called secondary

**Pull request merged**